### PR TITLE
Fixbug/mac network extension

### PR DESCRIPTION
**Pull request by AnhTVc**
Last commit of MAC NE from dev branch

### CODE REVIEW

File: client/ui/notificationhandler.cpp [hunk 1]
```diff
@@ -5,12 +5,15 @@
 #include <QDebug>
 #include "notificationhandler.h"
 
-#if defined(Q_OS_IOS)
+#if defined(Q_OS_IOS) || defined(MACOS_NE)
 #  include "platforms/ios/iosnotificationhandler.h"
+// #elif defined(MACOS_NE)
```
Comment (Nethius) [inline client/ui/notificationhandler.cpp]: please, cleanup

File: client/ui/systemtray_notificationhandler.h [hunk 2]
```diff
@@ -8,7 +8,7 @@
 #include "notificationhandler.h"
 
 #include <QMenu>
-#include <QSystemTrayIcon>
+#include <QSystemTrayIcon>'QMenu' file not found
```
Comment (Nethius) [inline client/ui/systemtray_notificationhandler.h]: what is this line of code for?

File: client/platforms/ios/PacketTunnelProvider+OpenVPN.swift [hunk 3]
```diff
@@ -47,7 +47,7 @@ extension PacketTunnelProvider {
         let configuration = OpenVPNConfiguration()
         configuration.fileContent = ovpnConfiguration
         if str.contains("cloak") {
-            configuration.setPTCloak()
+//            configuration.setPTCloak()
```
Comment (Nethius) [inline client/platforms/ios/PacketTunnelProvider+OpenVPN.swift]: do we need to delete this line?

File: client/macos/scripts/openvpn.sh [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+# Append macOS-specific build directory configurations to amnezia.xcconfig
+cat << EOF >> $WORKINGDIR/3rd/OpenVPNAdapter/Configuration/amnezia.xcconfig
+PROJECT_TEMP_DIR = $WORKINGDIR/3rd/OpenVPNAdapter/build/OpenVPNAdapter.build
+CONFIGURATION_BUILD_DIR = $WORKINGDIR/3rd/OpenVPNAdapter/build/Release-macos
+BUILT_PRODUCTS_DIR = $WORKINGDIR/3rd/OpenVPNAdapter/build/Release-macos
+EOF
+
+# Exclude UIKit, include Cocoa for macOS
+# echo "OTHER_LDFLAGS = -framework Cocoa" >> $WORKINGDIR/3rd/OpenVPNAdapter/Configuration/amnezia.xcconfig
```
Comment (Nethius) [inline client/macos/scripts/openvpn.sh]: please, cleanup

File: client/macos/networkextension/WireGuardNetworkExtension-Bridging-Header.h [hunk 5]
```diff
@@ -23,3 +19,8 @@ bool key_from_hex(uint8_t key[WG_KEY_LEN], const char* hex);
 bool key_eq(const uint8_t key1[WG_KEY_LEN], const uint8_t key2[WG_KEY_LEN]);
 
 void write_msg_to_log(const char* tag, const char* msg);
+
+// Khai báo hàm C để Swift có thể sử dụng
```
Comment (Nethius) [inline client/macos/networkextension/WireGuardNetworkExtension-Bridging-Header.h]: please, translate to english please

File: client/macos/networkextension/WireGuardNetworkExtension-Bridging-Header.h [hunk 6]
```diff
@@ -1,10 +1,6 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (Nethius) [inline client/macos/networkextension/WireGuardNetworkExtension-Bridging-Header.h:1]: please, return license description

File: client/CMakeLists.txt [hunk 7]
```diff
@@ -354,9 +378,12 @@ endif()
 if(IOS)
     include(cmake/ios.cmake)
     include(cmake/ios-arch-fixup.cmake)
-elseif(APPLE AND NOT IOS)
-    include(cmake/osxtools.cmake)
+elseif(APPLE AND NOT IOS AND NOT DEFINED MACOS_NE)
+    # include(cmake/osxtools.cmake)
```
Comment (Nethius) [inline client/CMakeLists.txt:385]: please, cleanup

File: client/ui/notificationhandler.cpp (hunk 1 above)
Comment (AnhTVc) [reply client/ui/notificationhandler.cpp]: Done

File: client/ui/systemtray_notificationhandler.h (hunk 2 above)
Comment (AnhTVc) [reply client/ui/systemtray_notificationhandler.h]: Done, remove it

File: client/macos/networkextension/WireGuardNetworkExtension-Bridging-Header.h (hunk 5 above)
Comment (AnhTVc) [reply client/macos/networkextension/WireGuardNetworkExtension-Bridging-Header.h]: done

File: client/macos/scripts/openvpn.sh (hunk 4 above)
Comment (AnhTVc) [reply client/macos/scripts/openvpn.sh]: done

File: client/macos/networkextension/WireGuardNetworkExtension-Bridging-Header.h (hunk 6 above)
Comment (AnhTVc) [reply client/macos/networkextension/WireGuardNetworkExtension-Bridging-Header.h:1]: done

File: client/ui/notificationhandler.cpp (hunk 1 above)
Comment (rasolAAaa) [reply client/ui/notificationhandler.cpp]: سلام خسته نباشید این خوبماراازافتاد لطف فرمایید کمک کنید تا اندازی شود من درشهردوراز بچه هام زندگی مکنم تنها برنامه بود که در این فیلتر بودن بازن بچه هام تصوری صحبت میکردم لطف فرمایید کمک کنید باتشکر فراوان

File: client/CMakeLists.txt (hunk 7 above)
Comment (AnhTVc) [reply client/CMakeLists.txt:385]: Done

File: client/platforms/ios/PacketTunnelProvider+OpenVPN.swift (hunk 3 above)
Comment (AnhTVc) [reply client/platforms/ios/PacketTunnelProvider+OpenVPN.swift]: fixed, we need setPTCloak() function

File: .gitmodules.swp
Comment (Nethius) [inline .gitmodules.swp:1]: do we need this file?

File: client/CMakeLists.txt [hunk 8]
```diff
@@ -47,18 +48,23 @@ set(LIBS ${LIBS}
     Qt6::Core5Compat Qt6::Concurrent
 )
 
-if(IOS)
+if(IOS OR MACOS_NE)
     set(LIBS ${LIBS} Qt6::Multimedia)
 endif()
+# message("Client desktop build ", ${MACOS_NE})
```
Comment (Nethius) [inline client/CMakeLists.txt:55]: please, cleanup

File: client/cmake/macos_ne.cmake [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+target_include_directories(${PROJECT} PRIVATE ${CMAKE_CURRENT_LIST_DIR})
+target_compile_options(${PROJECT} PRIVATE
+    -DGROUP_ID=\"${BUILD_IOS_GROUP_IDENTIFIER}\"
+    -DVPN_NE_BUNDLEID=\"${BUILD_IOS_APP_IDENTIFIER}.network-extension\"
+)
+
+set(WG_APPLE_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/3rd/amneziawg-apple/Sources)
+
+target_sources(${PROJECT} PRIVATE
+#    ${CMAKE_CURRENT_SOURCE_DIR}/platforms/ios/iosvpnprotocol.swift
```
Comment (Nethius) [inline client/cmake/macos_ne.cmake]: do we need this line?

File: client/macos/networkextension/CMakeLists.txt [hunk 10]
```diff
@@ -0,0 +1,208 @@
...
+    message("Add macros for MacOS Network Extension")
+else()
+    message("MACOS_NE is OFF")
+endif()
+
+set_target_properties(networkextension PROPERTIES
+    XCODE_PRODUCT_TYPE com.apple.product-type.app-extension
+    BUNDLE_EXTENSION appex
+
+    #MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_BINARY_DIR}/Info.plist
```
Comment (Nethius) [inline client/macos/networkextension/CMakeLists.txt]: do we need this line?

File: client/macos/networkextension/CMakeLists.txt [hunk 11]
```diff
@@ -0,0 +1,208 @@
...
+set_target_properties("networkextension" PROPERTIES
+    XCODE_ATTRIBUTE_DEVELOPMENT_TEAM "X7UJ388FXK"
+)
+
+find_library(FW_ASSETS_LIBRARY AssetsLibrary)
+find_library(FW_MOBILE_CORE MobileCoreServices)
+find_library(FW_UI_KIT UIKit)
+find_library(FW_LIBRESOLV libresolv.9.tbd)
+
+# set(OpenVPNAdapter_DIR "${CLIENT_ROOT_DIR}/3rd/")
```
Comment (Nethius) [inline client/macos/networkextension/CMakeLists.txt]: do we need this commented-out block of code?

File: client/macos/networkextension/CMakeLists.txt [hunk 12]
```diff
@@ -0,0 +1,208 @@
...
+
+# Setting the framework search paths using CLIENT_ROOT_DIR
+set(CMAKE_XCODE_ATTRIBUTE_FRAMEWORK_SEARCH_PATHS "${CLIENT_ROOT_DIR}/3rd/OpenVPNAdapter/build/Release-macos")
+
+# Linking the framework using CLIENT_ROOT_DIR
+target_link_libraries("networkextension" PRIVATE "${CLIENT_ROOT_DIR}/3rd/OpenVPNAdapter/build/Release-macos/OpenVPNAdapter.framework")
+
+
+
+#target_link_libraries(networkextension PRIVATE ${FW_ASSETS_LIBRARY})
```
Comment (Nethius) [inline client/macos/networkextension/CMakeLists.txt]: do we need this commented-out block of code?

File: deploy/build_macos_ne.sh [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+macos_main_uuid=$(grep UUID -A1 -a ~/Library/MobileDevice/Provisioning\ Profiles/AnhTVMacOSMain.provisionprofile | grep -io "[-A-F0-9]\{36\}")
+mv ~/Library/MobileDevice/Provisioning\ Profiles/AnhTVMacOSMain.provisionprofile ~/Library/MobileDevice/Provisioning\ Profiles/$macos_main_uuid.mobileprovision
+
+# Setup provisioning profile cho Network Extension (NE)
+echo "Setting up provisioning profile for Network Extension"
+cp $PROJECT_DIR/deploy/AnhTVMacOSNE.mobileprovision ~/Library/MobileDevice/Provisioning\ Profiles/macos_ne.mobileprovision
+macos_ne_uuid=$(grep UUID -A1 -a ~/Library/MobileDevice/Provisioning\ Profiles/macos_ne.mobileprovision | grep -io "[-A-F0-9]\{36\}")
+mv ~/Library/MobileDevice/Provisioning\ Profiles/macos_ne.mobileprovision ~/Library/MobileDevice/Provisioning\ Profiles/$macos_ne_uuid.mobileprovision
+
+# Giải mã và cài đặt chứng chỉ ký code
```
Comment (Nethius) [inline deploy/build_macos_ne.sh]: please, translate to english

File: deploy/build_macos_ne.sh [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+echo "Installing signing certificates..."
+echo $MAC_TRUST_CERT_BASE64 | base64 --decode > mac_trust_cert.pem
+echo $MAC_SIGNING_CERT_BASE64 | base64 --decode > mac_signing_cert.p12
+
+# Cài đặt chứng chỉ vào keychain
```
Comment (Nethius) [inline deploy/build_macos_ne.sh]: please, translate to english

File: deploy/build_macos_ne.sh [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+security create-keychain -p password build.keychain
+security default-keychain -s build.keychain
+security unlock-keychain -p password build.keychain
+security import mac_trust_cert.pem -k build.keychain -A
+security import mac_signing_cert.p12 -k build.keychain -P $MAC_SIGNING_CERT_PASSWORD -A
+
+# Thiết lập keychain cho quá trình ký
```
Comment (Nethius) [inline deploy/build_macos_ne.sh]: please, translate to english

File: .gitmodules.swp
Comment (AnhTVc) [reply .gitmodules.swp:1]: removed

File: client/CMakeLists.txt (hunk 8 above)
Comment (AnhTVc) [reply client/CMakeLists.txt:55]: Yes done

File: client/cmake/macos_ne.cmake (hunk 9 above)
Comment (AnhTVc) [reply client/cmake/macos_ne.cmake]: deleted

File: client/macos/networkextension/CMakeLists.txt (hunk 10 above)
Comment (AnhTVc) [reply client/macos/networkextension/CMakeLists.txt]: deleted

File: client/macos/networkextension/CMakeLists.txt (hunk 11 above)
Comment (AnhTVc) [reply client/macos/networkextension/CMakeLists.txt]: delete

File: client/macos/networkextension/CMakeLists.txt (hunk 12 above)
Comment (AnhTVc) [reply client/macos/networkextension/CMakeLists.txt]: formated

File: deploy/build_macos_ne.sh (hunk 14 above)
Comment (AnhTVc) [reply deploy/build_macos_ne.sh]: done

File: deploy/build_macos_ne.sh (hunk 15 above)
Comment (AnhTVc) [reply deploy/build_macos_ne.sh]: done

**Pull request closed without merging**